tests/LennardJonesForceTest.cpp

```diff
@@ -0,0 +1,78 @@
+#include <gtest/gtest.h>
+#include "Particle.h"
+#include "ParticleContainer.h"
+#include "GravitationalForce.h"
+#include <vector>
+#include <iostream>
+#include "FileReader.h"
+#include "utils/ArrayUtils.h"
+#include <string>
+#include <spdlog/spdlog.h>
+#include <spdlog/sinks/basic_file_sink.h>
+#include <spdlog/fmt/ostr.h>
+#include "LennardJonesForce.h"
+#include "Force.h"
+#include "Logger.h"
+
+
+class LennardJonesForceTest : public ::testing::Test {
+protected:
+    std::vector<Particle> particles;
+    ParticleContainer *pc;
+    char* testfile = const_cast<char*>("../tests/test_cases/two_body.txt");
+
+
+    void SetUp() override {
+        FileReader fileReader;
+        fileReader.readFile(particles, testfile);
+        pc = new ParticleContainer(particles, new LennardJonesForce(),true);
+        spdlog::set_level(spdlog::level::info);
+        test_logger -> info("Particle Container created");
+    }
+
+    void TearDown() override {
+        test_logger->info("Particle Container deleted\n\n");
+        delete pc;
```

Comment:
with smart pointers, you wouldn't need to take care of this.

Reply:
Sorry I don't really understand what you mean by smart pointer (c++ is completely new for me), could you please give us a hint about this?

Reply:
Smart pointers take care of memory allocation/reallocation automatically (so you don't need to explicitly **delete** a pointer to free the memory). We briefly mentioned it in the presentation for worksheet 1. I provided links in one of the other comments. Have a look at that. Basically, as program gets complex, it becomes harder to keep track of when a pointer goes out of scope without freeing the memory, smart pointers take care of this. And in modern C++, we almost never use raw pointers now, so not only it might become critical at some point, it is also a good practice in general!

Reply:
Thank you